OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -282,6 +282,12 @@ public interface IMove
 		bool CanEnterTargetNow(Actor self, Target target);
 	}
 
+	public interface ITemporaryBlocker
+	{
+		bool CanRemoveBlockage(Actor self, Actor blocking);
```

Comment:
The interface suggests the actor is a _temporary_ blocker, `CanRemoveBlockage` will return different results for different `blocking` actors or when powered down etc. This is essentially how gates work, the pathfinder will use this function to see if the unit can potentially remove the blockage and if yes will just "bump" into it.